src/passage_flex.rs

```diff
@@ -43,373 +37,27 @@ impl PassageFlex {
-    ///
-    /// let external_id = "00000000-0000-0000-0000-000000000001";
-    /// let last_year = Utc::now().naive_utc().date() - Duration::days(365);
-    ///
-    /// let passkey_devices = passage_flex.get_devices(external_id.to_string()).await.unwrap();
-    ///
-    /// for device in passkey_devices {
-    ///     let last_login_at_parsed =
-    ///         NaiveDate::parse_from_str(&device.last_login_at, "%Y-%m-%dT%H:%M:%S%z").unwrap();
-    ///
-    ///     if last_login_at_parsed < last_year {
-    ///         if let Err(err) = passage_flex
-    ///             .revoke_device(external_id.clone(), device.id)
-    ///             .await
-    ///         {
-    ///             // device couldn't be revoked
-    ///         }
-    ///     }
-    /// }
-    /// ```
-    pub async fn revoke_device(&self, external_id: String, device_id: String) -> Result<(), Error> {
-        let user_id = self.get_id(external_id).await?;
-        user_devices_api::delete_user_devices(&self.configuration, &user_id, &device_id)
-            .await
-            .map_err(Into::into)
-    }
-
-    /// Retrieves information about a user by their user ID in Passage.
-    ///
-    /// # Arguments
-    ///
-    /// * `user_id` - The ID of the user in Passage.
-    ///
-    /// # Returns
-    ///
-    /// A `Result` containing the `UserInfo` struct or an `Error`.
-    ///
-    /// # Examples
-    ///
-    /// ```ignore
-    /// use passage_flex::PassageFlex;
-    ///
-    /// let passage_flex = PassageFlex::new(
-    ///     std::env::var("PASSAGE_APP_ID").unwrap(),
-    ///     std::env::var("PASSAGE_API_KEY").unwrap(),
-    /// );
-    ///
-    /// let user_id = "user_id_string";
-    /// let user_info = passage_flex.get_user_by_id(user_id.to_string()).await.unwrap();
-    /// println!("{:?}", user_info.external_id);
-    /// ```
-    pub async fn get_user_by_id(&self, user_id: String) -> Result<Box<UserInfo>, Error> {
-        users_api::get_user(&self.configuration, &user_id)
-            .await
-            .map(|response| {
-                Box::new(UserInfo {
-                    created_at: response.user.created_at,
-                    external_id: response.user.external_id,
-                    id: response.user.id,
-                    last_login_at: response.user.last_login_at,
-                    login_count: response.user.login_count,
-                    status: response.user.status,
-                    updated_at: response.user.updated_at,
-                    user_metadata: response.user.user_metadata,
-                    webauthn: response.user.webauthn,
-                    webauthn_devices: response.user.webauthn_devices,
-                    webauthn_types: response.user.webauthn_types,
-                })
-            })
-            .map_err(Into::into)
+        self.user.configuration.base_path = format!("{}/v1/apps/{}", server_url, self.app_id);
+        self.auth.configuration.base_path = format!("{}/v1/apps/{}", server_url, self.app_id);
```

Comment:
necessary to mock the server calls in the unit tests. this can be removed once the tests are moved to the e2e suite